lib/kitchen/provisioner/_states/mock.py

```diff
@@ -34,40 +50,53 @@ def _mock_get(tgt, fun, tgt_type='glob', exclude_minion=False, expr_form=None):
         return {}
     # TODO(ppg): allow lookup based on arg too
     d = d[fun]
-    return { tgt: d }
+    return {tgt: d}
+
 
 def remote_functions(name):
     global remote_functions_data
     # FIXME(ppg): drive via configuration
     for file in glob.glob('/tmp/kitchen/srv/remote_functions/*'):
         remote_functions_data.update(yaml.load(open(file)))
     __salt__['publish.publish'] = _mock_publish
-    return { 'name': 'remote_functions', 'changes': {}, 'result': True, 'comment': 'Replaced publish.publish with mock' }
+    return {'name': 'remote_functions', 'changes': {}, 'result': True, 'comment': 'Replaced publish.publish with mock'}
+
 
 def _mock_publish(tgt, fun, arg=None, tgt_type='glob', returner='', timeout=5, via_master=None, expr_form=None):
     global remote_functions_data
     log.info('MOCK: Publishing {0!r} for {1}'.format(fun, tgt))
+    log.debug('MOCK: arg type: {0}'.format(type(arg)))
+    log.debug('MOCK: arg: {0!r}'.format(arg))
     kwargs = salt.utils.args.yamlify_arg(arg)
+    log.debug('MOCK: kwargs type: {0}'.format(type(kwargs)))
+    log.debug('MOCK: kwargs: {0}'.format(kwargs))
 
     # Special case some functions for convinient usage
     # TODO(ppg): allow custom python files provided to override a function
     #   mock_remote_functions:
     #     'x509.sign_remote_certificate': mock_sign_remote_certificate.py
     if fun == 'x509.sign_remote_certificate':
+        # if salt.utils.args.yamlify failed to make a dict, try ast
+        if not isinstance(kwargs, dict):
+            kwargs = ast.literal_eval(kwargs)
```

Comment:
Is there a harm to doing this up where `kwargs` is set first? I.e. can we always fallback to `ast` and not just for `x509.sign_remote_certifcate`?

Reply:
We could do `ast.literal_eval` for everything, but I thought that modifying as few functions as possible would be better in the long run to limit how many things the mock changes.